### ISSUE TITLE
Item Statistics

### ISSUE THREAD

**Issue by alexalperov** [yasp-dota/yasp#731]
Dotamax has a nice feature which documents hero advantages/disadvantages" in certain matchups. For example:
![image](https://cloud.githubusercontent.com/assets/15504259/10861661/8a9ec25c-7f56-11e5-9932-316597da8245.png)

Something which has not been implemented anywhere is a similar application for items. For example:

Wraith King
_________

Wraith King's Best Versus

Item           Anti-Index       Win Rate
Daedalus       +6.8%           53%

Wraith King's Worst Versus

Item           Anti-Index       Win Rate
Diffusal Blade   -12.4%        49%
Scythe             -4.4%         47%

With Yasp's in-depth knowledge of item purchase items, etc it would allow even greater flexibility. You can create scales that show the effectiveness of the items at different points in the game. 

I understand one of the limits of Yasp is that it does not parse as much game data as dotamax and dotabuff, but this feature would give a distinct competitive edge over other statistics websites as well as provide a strategic benefit for DOTA players. 

Thoughts?

**Comment by howardchung** [comment]
Interesting.

So you'd sum together all of the opponent's items ever bought, and add a win/loss data point for that match for that hero/item combination?

**Comment by alexalperov** [comment]
Yeah that's exactly what I would do. In the end you would have data sets for every hero's winrate against every single item.

**Comment by howardchung** [comment]
concerns:

* How to store this data?  Create a new SQL relation for it?  What would the schema look like?

* What is the cost of adding together all of a team's items?  I suppose this could be done once per team, then 10 data points created and the backing data store updated.

**Comment by albertcui** [comment]
This would probably be a good task for a graph based storage backend. A
background task could run this on an interval. I don't think it needs to be
updated in real time.

**Comment by howardchung** [comment]
You want to play with it?  I think this is where the "fan-out" stuff starts
happening.  When we add a match we're getting to the point where we trigger
a lot of updates and we need to do this asynchronously so we don't stop
importing new matches.  Updating player caches is already the bottleneck in
some cases.  I thought about modifying scanner.js to just put the API
results in a queue, then create a new worker type that handles
inserting/fan-out tasks.  We could use multiple threads to speed it up.
